internal/cache/cache.go

```diff
@@ -35,6 +35,11 @@ type Cache struct {
 // New returns a new Cache that will cache items for
 // at most maxAge.
 func New(maxAge time.Duration) *Cache {
+	// A maxAge of 1ns would cause the expire calculations to panic.
+	// If that age is specified increase it to 2ns as nothing is likely to notice.
+	if maxAge == time.Nanosecond {
```

Comment:
because rand.Int63n() panics if it's input is <= 0 (https://golang.org/pkg/math/rand/#Int63n) It would also panic on 0ns, so I should probably make that test a bit wider.